amplify/backend/function/fetchDomainBalance/src/index.js

```diff
@@ -0,0 +1,61 @@
+require('cross-fetch/polyfill');
+
+const {
+    utils: { Logger },
+} = require('ethers');
+const { getMonthsFromNow, getDaysFromNow, getMonthFormat } = require('./utils');
+const ExchangeRatesFactory = require('./config/exchangeRates');
+const { getInOutActions, getTokensDatesMap, filterActionsWithinTimeframe, updateMonthlyBalance, update30DaysBalance } = require('./services/actions');
+const { getTotalFiatAmountFor } = require('./services/tokens');
+Logger.setLogLevel(Logger.levels.ERROR);
+
+// @TODO maybe rename the lambda function
+
+/**
+ * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
+ */
+exports.handler = async (event) => {
+    try {
+        const { colonyAddress, domainAddress, chainId, selectedCurrency } = event.arguments?.input || {};
+        const fourMonthsFromNow = getMonthsFromNow(4);
+        const last30Days = getDaysFromNow(30);
+
+        const inOutActions = await getInOutActions(colonyAddress, domainAddress);
+        const inOutActionsWithinTimeframe = filterActionsWithinTimeframe(inOutActions, fourMonthsFromNow);
+        const timeframeMonthlyBalance = updateMonthlyBalance(inOutActionsWithinTimeframe, domainAddress);
+
+        const last30DaysActions = filterActionsWithinTimeframe(inOutActions, last30Days);
+        const last30DaysBalance = update30DaysBalance(last30DaysActions, domainAddress);
+
+        const exchangeRates = await ExchangeRatesFactory.getExchangeRates(getTokensDatesMap(inOutActionsWithinTimeframe), selectedCurrency, chainId);
+
+        const inOutTimeframeMontlyBalance = {};
+
+        for (let month in timeframeMonthlyBalance) {
+            const balance = timeframeMonthlyBalance[month];
+            const totalIn = await getTotalFiatAmountFor(balance.in, exchangeRates);
+            const totalOut = await getTotalFiatAmountFor(balance.out, exchangeRates);
+            inOutTimeframeMontlyBalance[month] = {
+                totalIn,
+                totalOut,
+            };
+        }
+
+        const last30DaysBalanceTotalIn = await getTotalFiatAmountFor(last30DaysBalance.in, exchangeRates);
+        const last30DaysBalanceTotalOut = await getTotalFiatAmountFor(last30DaysBalance.out, exchangeRates);
+
+        return {
```

Comment:
Need to re-check how granular we want the response to be -> maybe implement a helper to retrieve data on a daily/weekly/monthly basis